### PR TITLE
Framework: remove `wpcom-unpublished` dependency

### PR DESCRIPTION
**Pull request by retrofox**
[`wpcom-unpublished`](https://github.com/Automattic/wpcom-unpublished) is a library used in calypso to make requests to REST-API. It inherits from [wpcom.js](https://github.com/Automattic/wpcom.js) and it was needed in the first stage of the calypso development.
Quite the contrary It does development work and more complicated maintenance.

So to remove this lib I've implemented its methods directly in wpcom.js. Affected methods:

- [x] Me#postsList()
  * Same method already [defined in wpcom@4.9.1](https://github.com/Automattic/wpcom.js/blob/master/lib/site.js#L216).

- [x] Site#postCounts()
  * Using [WPComUndocummented#postCounts()](https://github.com/Automattic/wp-calypso/blob/master/client/lib/wpcom-undocumented/lib/site.js#L186) method.

- [x] Site#postTypesList()
  * Same method already [defined in wpcom@4.9.1](https://github.com/Automattic/wpcom.js/blob/master/lib/runtime/site.get.json#L12) in the runtime building process.

- [x] Site#primaryDomain()
  * Replace method defined in `wpcom-undocumented` -> f7f6d001983c1ec5e0e0b5b8c084eafb37c5b0fd

_before_
```es6
wpcom.undocumented().getPrimaryDomain( siteId, onComplete ) {
  // ...
} );
```

_after_
```es6
wpcom
  .site( siteId )
  .domain()
  .getPrimary( onComplete ) {
    // ...
  } );
```

- [x] Site#pluginsList()
  * Same method already [defined in wpcom@4.9.1](https://github.com/Automattic/wpcom.js/blob/master/lib/runtime/site.get.json#L9) in the runtime building process.
- [x] Site#wordAds()
  * `site.wordAds()` has been implemented in the same way in `wpcom.js`

- [x] Post#subscribers()
  * PR: https://github.com/Automattic/wpcom.js/pull/167

- [x] Post#subscriber()
  * PR: https://github.com/Automattic/wpcom.js/pull/167

- [x] Subscriber#mine()
  * PR: https://github.com/Automattic/wpcom.js/pull/167

- [x] Subscriber#state()
  * PR: https://github.com/Automattic/wpcom.js/pull/167

- [x] Subscriber#add()
  * PR: https://github.com/Automattic/wpcom.js/pull/167

- [x] Subscriber#del()
  * PR: https://github.com/Automattic/wpcom.js/pull/167

- [x] Subscriber#delete()
  * PR: https://github.com/Automattic/wpcom.js/pull/167

- [x] SiteWordAds#settings()
- [x] SiteWordAds#earnings()
- [x] SiteWordAds#tos()

- [x] SiteWordAdsSettings#get()
- [x] SiteWordAdsSettings#update()

- [x] SiteWordAdsEarnings#get()

- [x] SiteWordAdsTos#get()
- [x] SiteWordAdsTos#update()
- [x] SiteWordAdsTos#sign()

### Testing

Check and test the pages / sections that are already using some of these methods. Everything should be ok.

  * **Site#postsList**:
    * `client/lib/menu-data/menu-data.js`
    * `client/lib/posts/actions.js`
    * `client/state/posts/actions.js:`

  * **Site#postTypesList**:
    * `client/lib/post-types-list/list.js`
    * `client/state/post-types/actions.js`

  * **Site#primaryDomain**:
    * `client/lib/domains/index.js`

  * **Site#wordAds()**:
    * `client/lib/ads/actions.js`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -113,6 +113,7 @@
     "walk": "2.3.4",
     "webpack": "1.12.6",
     "webpack-dev-middleware": "1.2.0",
+    "wpcom": "4.9.1",
```
Comment (nylen) [inline package.json]: `wpcom` is not a dependency until now.  Why do we add it here?

File: package.json [hunk 2]
```diff
@@ -113,6 +113,7 @@
...
     "wpcom-proxy-request": "1.0.5",
     "wpcom-unpublished": "1.0.10",
```
Comment (nylen) [inline package.json]: Can we remove `wpcom-unpublished` yet?

Comment (retrofox) [inline package.json]: This PR is `In Progress` status yet but that's the idea tough.

**Comment by retrofox** [comment]
cc @rralian @nylen @timmyc

**Comment by timmyc** [comment]
@retrofox, first of all :bow: - thank you for doing this.  The one item above I'm not sure how to test is "Site#Site#wordAds():".  In order to test this via the UI, do I need to have a word ads enabled site?

**Comment by retrofox** [comment]
> ... The one item above I'm not sure how to test is "Site#wordAds():". In order to test this via the UI, do I need to have a word ads enabled site?

mmm I'm not completely sure how to test this stuff but I think you're right. Let's ask how to do it. I've tested through wpcom.js tests.

**Comment by timmyc** [comment]
@retrofox tested out the ad section and also verified all is flowing to WP-API endpoints as expected.

LGTM

**Comment by dbspringer** [comment]
👌

**Comment by retrofox** [comment]
Thanks guys for the review

**Pull request merged**